### PR TITLE
build: Create Vue 3 migration test package

### PR DESCRIPTION
**Pull request by alvarodE**
[EMP-3849](https://searchbroker.atlassian.net/browse/EMP-3849)

## Motivation and context

Add a new private package to test migrated components with the [Vue 3 migration build](https://v3-migration.vuejs.org/migration-build.html).

## What is the destination branch of this PR?
- [x] `Main`

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

[EMP-3849]: https://searchbroker.atlassian.net/browse/EMP-3849?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/_vue3-migration-test/package.json
Comment (diegopf) [inline packages/_vue3-migration-test/package.json:1]: Running the dev server it works but I get an error message from vite telling me that ` @empathyco/x-components` and ` @empathyco/x-components/js/components/animations/animate-width.vue` could not be resolved. Would be posible to add `@empathyco/x-components` as workspace dependency? I've tried do it with 
```
pnpm add @empathyco/x-components
```
and seems that is working just fine afterwards. If that's an issue then I think we need to exclude them from the [optimizeDeps](https://vitejs.dev/config/dep-optimization-options#optimizedeps-exclude) option in vite.

**Comment by diegopf** [comment]
Really nice job! 👏

File: packages/_vue3-migration-test/package.json
Comment (alvarodE) [reply packages/_vue3-migration-test/package.json:1]: I've done the change on Vite as my intention with this new package is to use x-components source files as if they were part of the new package 😀

**Review by diegopf — DISMISSED** [review]

File: packages/_vue3-migration-test/src/components/animations/test-animate-width.vue [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <div>
+    <button @click="visible = !visible">Toggle</button>
+    <AnimateWidth>
+      <div v-if="visible" style="width: 300px">Element to animate</div>
+    </AnimateWidth>
+  </div>
+</template>
+
+<script lang="ts">
```
Comment (joseacabaneros) [inline packages/_vue3-migration-test/src/components/animations/test-animate-width.vue:10]: You can use `script setup` here if you want
```
<script setup lang="ts">
  import { ref } from 'vue';
  import AnimateWidth from '../../../../x-components/src/components/animations/animate-width.vue';

  const visible = ref(true);
</script>
```

Also changing it, this TS error is solved
Screenshot 2024-04-18 at 6 37 33 PM

**Review by joseacabaneros — DISMISSED** [review]
Good starting point! 🔝 
Tested with `serve` and `build` and works as expected

**Review by diegopf — APPROVED** [review]

**Review by joseacabaneros — APPROVED** [review]

**Pull request merged**